test/util/test_log.py

```diff
@@ -33,3 +36,72 @@ def test_LoggingContext():
     # check logging levels are as they were before
     assert idx_logger.getEffectiveLevel() == logging.DEBUG
     assert dials_logger.getEffectiveLevel() == logging.DEBUG
+
+
+def test_easy_mp_logging():
+    """
+    Check the filtering of log records from easy_mp child processes.
+
+    Child processes spawned/forked with libtbx.easy_mp use a logger that caches the
+    raised messages with dials.util.log.CacheHandler, rather than emitting them.
+    Here we check that the 'dials' loggers in the child processes inherit the threshold
+    logging level of the 'dials' logger in the parent process.
+    """
+    test_log_message = "Here's a test log message."
+
+    def log_something(_: Any) -> List[logging.LogRecord]:
+        """
+        Create a dummy info log record.
+
+        A little dummy function to pass to the dials.util.mp parallel map functions,
+        which simply logs a single info message.
+
+        Args:
+            _:  Absorb the expected argument and do nothing with it.
+
+        Returns:
+            The log records created during the calling of this function.
+        """
+        dials.util.log.config_simple_cached()
+        logger = logging.getLogger("dials")
+        logger.info(test_log_message)
+        (handler,) = logger.handlers
+        return handler.records
+
+    # Check that the child processes cache INFO log messages when the parent process
+    # logger's threshold level is the default (DEBUG).
+    dials.util.log.config(verbosity=2)
```

Comment:
This (and other statements) will leak into the next test.

Reply:
Yes, I realised this as I was committing it but I wanted to get some CI checks for the benefit of my own understanding of things, so I pushed it like this and intend to tidy it up later.  Hence draft.